### PR TITLE
net: lib: nrf_cloud: add mutex around modem info struct

### PR DESCRIPTION
**Pull request by jayteemo**
IRIS-6938

### CODE REVIEW

**Review by plskeggs — APPROVED** [review]
The error goes away now?

**Comment by jayteemo** [comment]
> The error goes away now?

Yes, but feel free to test with this change to confirm.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c [hunk 1]
```diff
@@ -1650,13 +1659,17 @@ int nrf_cloud_modem_info_json_encode(const struct nrf_cloud_modem_info *const mo
...
-			LOG_ERR("modem_info_params_get() failed: %d", err);
+			LOG_ERR("get_modem_info() failed: %d", err);
 			goto cleanup;
 		}
+		(void)k_mutex_lock(&modem_inf_mutex, K_FOREVER);
 		mpi = &modem_inf;
 	}
 
 	err = encode_modem_info_json_object(mpi, tmp, mod_inf->application_version);
+	if (mpi == &modem_inf) {
```
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:1670]: can `mod_inf->mpi` definitely never already be equal to `&modem_inf`?

Maybe a dedicated boolean that gets set true prior to the lock would be a good idea just to sidestep even having to answer that question

Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c]: > would be a good idea just to sidestep even having to answer that question

🤔 
ok... flag added, discussion side-stepped.

**Review by glarsennordic — APPROVED** [review]

**Pull request merged**